src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs

```diff
@@ -183,26 +185,36 @@ void Run (bool useMarshalMethods)
 			}
 
 			// Now that "never" never happened, we can proceed knowing that at least the assembly sets are the same for each architecture
-			var nativeCodeGenStates = new Dictionary<AndroidTargetArch, NativeCodeGenState> ();
-			bool generateJavaCode = true;
+			var nativeCodeGenStates = new ConcurrentDictionary<AndroidTargetArch, NativeCodeGenState> ();
 			NativeCodeGenState? templateCodeGenState = null;
 
-			foreach (var kvp in allAssembliesPerArch) {
+			var firstArch = allAssembliesPerArch.First ().Key;
+			var generateSucceeded = true;
+
+			// Generate Java sources in parallel
+			Parallel.ForEach (allAssembliesPerArch, (kvp) => {
```

Comment:
Does this use Mono.Cecil in parallel? We found problems using Mono.Cecil on different threads in:
* https://github.com/dotnet/java-interop/pull/43
* https://github.com/dotnet/java-interop/pull/365

Random, weird things would happen.

Reply:
I believe we are fine here. While we are using the `Mono.Cecil` library in parallel, the instances running on not using the same assemblies or `AssemblyResolver`.

That is, one thread is generating all JCWs for `arm64` from the `arm64` assemblies:

- obj\Debug\net9.0-android\android\assets\arm64-v8a\Mono.Android.dll
- obj\Debug\net9.0-android\android\assets\arm64-v8a\Mono.Android.Runtime.dll
- obj\Debug\net9.0-android\android\assets\arm64-v8a\Mono.Android.Export.dll

while another is generating all JCWs from the `x86` assemblies:

- obj\Debug\net9.0-android\android\assets\x86_64\Mono.Android.dll
- obj\Debug\net9.0-android\android\assets\x86_64\Mono.Android.Runtime.dll
- obj\Debug\net9.0-android\android\assets\x86_64\Mono.Android.Export.dll

No `AssemblyResolver`, `AssemblyDefinition`, or `TypeDefinition` instances are being accessed from more than one thread at a time.

Reply:
Indeed, we should be fine here.  I designed the per-RID support so that, eventually, it can be parallelized.  All the caches etc are per-arch (and if anything isn't, then it's either read-only or a bug).